src/ir/item.rs

```diff
@@ -325,37 +325,37 @@ impl CanDeriveDefault for Item {
 
 impl CanDeriveCopy for Item {
     fn can_derive_copy(&self, ctx: &BindgenContext) -> bool {
-        self.id().can_derive_copy(ctx)
+        self.id().can_derive_copy(ctx) && !self.is_opaque(ctx, &())
     }
 }
 
 impl CanDeriveHash for Item {
     fn can_derive_hash(&self, ctx: &BindgenContext) -> bool {
-        self.id().can_derive_hash(ctx)
+        self.id().can_derive_hash(ctx) && !self.is_opaque(ctx, &())
```

Comment:
Instead of this, the right place for this is returning `CanDerive::No` here:

https://github.com/rust-lang/rust-bindgen/blob/ec85170cb1a58737e78d8d3bacddd8866795c175/src/ir/analysis/derive.rs#L158

Reply:
Yeah. I started there an then thought that maybe I should "attack" this at the source to make sure i'm not missing other places.
I'll change.

Reply:
Another problem is that this also affects `Default` and `Debug`
And for some reason also affected stuff the current code missed like https://github.com/rust-lang/rust-bindgen/blob/f6867510999e2010db1ee80ef3d63c4befc9881e/tests/expectations/tests/template.rs#L366

Reply:
Well, sure that'd affect more traits, you can choose what it affects conditionally adding something like `self.derive_trait.can_derive_opaque()`.